api/src/main/java/com/javadiscord/jdi/core/api/utils/CacheUpdater.java

```diff
@@ -0,0 +1,44 @@
+package com.javadiscord.jdi.core.api.utils;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+import com.javadiscord.jdi.internal.cache.Cache;
+
+public class CacheUpdater {
+
+    private final Cache cache;
+
+    public CacheUpdater(Cache cache) {
+        this.cache = cache;
+    }
+
+    public <T> void updateCache(T result) {
+        if (result == null) {
+            return;
+        }
+        try {
+            Field guildIdField = result.getClass().getDeclaredField("guildId");
+            Field idField = result.getClass().getDeclaredField("id");
+
+            long guildId = getLongFromField(guildIdField, result);
+            long id = getLongFromField(idField, result);
+
+            cache.getCacheForGuild(guildId).add(id, result);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            // NoSuchFieldException: don't cache result
+        }
+    }
+
+    public <T> void updateCache(List<T> resultList) {
+        resultList.forEach(this::updateCache);
+    }
+
+    private <T> long getLongFromField(Field field, T result) throws IllegalAccessException {
+        field.setAccessible(true);
+        if (field.getType() == String.class) {
```

Comment:
This could potentially throw an exception if the id is a string but also not something that can  be parsed to a Long for example the `id` is "test". I believe there is 1 model which has this exception but I don't remember from memory which one this is. Might be best to catch the error that `Long.parseLong` could thow with a `trace` level log statement.

Reply:
Added log tracing for this. I caught the exception up the stack so that caching does not halt entirely in the event of caching a list.

Reply:
Looks good, instead of `String.format` you can simply do `"Failed to parse %s=%s as a long".formatted(field.getName(), field.get(result));`

Reply:
It doesn't work that way. You either have to call String.format to use those placeholders, or use String concatenation. Pushed a commit throw the original exception.

Reply:
I meant something like this: https://github.com/javadiscord/java-discord-api/blob/adfd89b7880c49712c4875e23214823fe6e2c927/api/src/main/java/com/javadiscord/jdi/internal/api/channel/AddThreadMemberRequest.java#L12

What you have done now is also okay, everything LGTM!

Reply:
`.formatted`

Yes mb, I overlooked that call